services/ymax-planner/src/support.ts

```diff
@@ -441,6 +441,7 @@ type ScanOpts = {
   chunkSize?: number;
   log?: (...args: unknown[]) => void;
   signal?: AbortSignal;
+  chunkCallback?: (startBlock: number, endBlock: number) => void;
```

Comment:
Please rename to indicate that this callback is invoked only for chunks not containing a match.
Suggestion:
```suggestion
  onRejectedChunk?: (startBlock: number, endBlock: number) => Promise<void> | void;
```
(note also the change of return type)